### PR TITLE
Fix symbol deprecation

### PR DESCRIPTION
**Pull request by malmaud**

### CODE REVIEW

File: src/functors.jl [hunk 1]
```diff
@@ -37,7 +37,7 @@ macro functor2(F, fun, T)
 end
 
 default_functorsym(f::Symbol) =
-    (fstr = string(f); symbol(string(uppercase(fstr[1]), fstr[2:end], "Fun")))
+    (fstr = string(f); Symbol(string(uppercase(fstr[1]), fstr[2:end], "Fun")))
```
Comment (tkelman) [inline src/functors.jl:40]: `@compat` (plus a minimum version in REQUIRE) ?

**Comment by davidavdav** [comment]
I second that.

**Comment by tkelman** [comment]
does anyone active have commit access here? if not, packages that use this one should consider updating to avoid it. if that's not practical, we can consider redirecting metadata to point to a fork somewhere such as JuliaMath.

**Comment by yuyichao** [comment]
I do. And I can merge after the 0.3 issue is fixed.

**Comment by tkelman** [comment]
or 0.3 support could be dropped

**Comment by bicycle1885** [comment]
Perhaps we should deprecate this package and put a link to [StatsFuns.jl](https://github.com/JuliaStats/StatsFuns.jl) on README.

**Comment by malmaud** [comment]
Guess I'll just close this if this package is defunct now.

**Pull request closed without merging**